src/OpenTelemetry.Api/OpenTelemetry.Api.csproj

```
@@ -12,6 +12,10 @@
     <Nullable>disable</Nullable>
   </PropertyGroup>
 
+  <ItemGroup>
+    <Compile Include="$(RepoRoot)\src\OpenTelemetry\Internal\UnconditionalSuppressMessageAttribute.cs" LinkBase ="Common" Condition ="'$(TargetFrameworks)' != 'netstandard2.0'" />
```

```suggestion
    <Compile Include="$(RepoRoot)\src\OpenTelemetry\Internal\UnconditionalSuppressMessageAttribute.cs" LinkBase ="Common" Condition="'$(TargetFrameworkIdentifier)' != '.NETCoreApp'" />
```

Since this assembly only builds for `netstandard2.0;net462` you could also add it unconditionally:

```suggestion
    <Compile Include="$(RepoRoot)\src\OpenTelemetry\Internal\UnconditionalSuppressMessageAttribute.cs" LinkBase ="Common" />
```

---

Typically what we have done with these shims in the past is have a preprocessor directive in the file itself. Example: https://github.com/open-telemetry/opentelemetry-dotnet/blob/4ebbccfe7b16e65470f56b22c25d7a60561730a5/src/OpenTelemetry/Internal/Shims/NullableAttributes.cs#L22

Is there a benefit to doing it via conditions in the csproj link definition? One thing I like about having the directives in the file is it seems a bit more self-documenting.

---

I tried the preprocessor approach and it looks like this:
https://github.com/open-telemetry/opentelemetry-dotnet/pull/4428/files/9a629dcb5d3d3c5a20f6860d60beca86df9ab75b

Related discussion: https://github.com/open-telemetry/opentelemetry-dotnet/pull/4428/files/9a629dcb5d3d3c5a20f6860d60beca86df9ab75b#r1175852874

---

Plus, I guess we may need to leverage other TrimmingAttributes, which are `internal` to System.Runtime while working on addressing other trimming warnings.

https://github.com/AzureAD/azure-activedirectory-identitymodel-extensions-for-dotnet/blob/dev/src/Common/TrimmingAttributes.cs

---

> Is there a benefit to doing it via conditions in the csproj link definition?

(totally just my opinion) The .csproj is what defines which TFMs the project builds for. So when you add/remove/etc TargetFrameworks from the .csproj you can also see all the Conditions based on those TFMs. This works well for whole files. `#if` usually works better when it is contained to a small number of methods that need to switch based on TFM.